helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -217,7 +217,6 @@ public void assignWorkflow(String workflow, WorkflowConfig workflowCfg,
     } else {
       LOG.debug("Workflow {} is not ready to be scheduled.", workflow);
     }
-    _clusterDataCache.updateWorkflowContext(workflow, workflowCtx);
```

Comment:
This update is not necessary. When there is a job that needs to be scheduled, we update the workflow context there. If there is no job, then we do not need to update the workflow context